formats/json/commonMain/src/kotlinx/serialization/json/JsonElement.kt

```diff
@@ -61,6 +63,17 @@ public fun JsonPrimitive(value: Number?): JsonPrimitive {
     return JsonLiteral(value, isString = false)
 }
 
+/**
+ * Creates [JsonPrimitive] from the given number.
+ */
+public fun JsonLiteralNumber(value: String?): JsonPrimitive {
```

Comment:
@sandwwraith Regarding the rename, you suggested `JsonUnquotedPrimitive`. However, the provided value might not be a primitive (users could use it to encode JSON arrays or objects), and the because might have quotes it could be a bit misleading - the new function doesn't really care about quotes.

How about `JsonLiteralElement` or `JsonRawElement`?

Reply:
Well, the 'unquoted' part simply means that we won't put quotes around the value. If one creates `JsonUnquotedPrimitive("\"foo\"")` of course it would be quoted — but that is defined by the data, not the function behavior.

> However, the provided value might not be a primitive 

Yes, that's a valid point. Maybe `JsonUnquotedLiteral` then?

Reply:
I'd prefer having 'Raw' because it matches similar APIs in other serialization libraries, and it's a common word ('raw' is mentioned in each of the tickets I linked in the PR description), but that's minor - `JsonUnquotedLiteral` is fine.

I have another thought though. What about providing a function overload for `JsonPrimitive()`?

```kotlin
/**
 * Creates [JsonPrimitive] from the given string.
 *
 *  @param[quoted] If false, the value will not be wrapped in quotes <insert more docs here>
 */
@ExperimentalSerializationApi
public fun JsonPrimitive(value: String?, quoted: Boolean): JsonPrimitive {
    return if (!quoted) JsonPrimitive(value)
    else when (value) {
        null -> JsonNull
        JsonNull.content -> throw JsonEncodingException("It is impossible to create a literal unquoted value of 'null'. If you want to create JSON null literal, use JsonNull object, otherwise, use JsonPrimitive")
        else -> JsonLiteral(value, isString = false, coerceToInlineType = jsonRawElementDescriptor)
    }
}
```

It's a little odd, because `@ExperimentalSerializationApi` means it has to be a separate function rather than adding a new argument with a default to the existing `JsonPrimitive()`, but if this gets promoted to stable, then the function could be merged.


```kotlin
public fun JsonPrimitive(value: String?, quoted: Boolean = true): JsonPrimitive {
  // as above
}
```